### DISCUSSION TITLE
Support for M5Though

### DISCUSSION THREAD

**Discussion by ritterr** [HASwitchPlate/openHASP#426] [Hardware]
I was wondering if there is any chance to get the M5Tough hardware supported or get some directions on how to compile openHASP for the M5Tough. 

I have flashed the M5Stack Core2 firmware onto the M5Tough but eventually the display (same as the M5Stack Core2) remains black even though the firmware is booting but fails to load the config.json from the fs. I am aware the touch controller is different to the M5Stack Core2.

Any advise would be much appreciated and I am more than happy to donate a device.

        open____ _____ _____ _____
          |  |  |  _  |   __|  _  |
          |     |     |__   |   __|
          |__|__|__|__|_____|__|
        Home Automation Switch Plate
        Open Hardware edition v0.6.3

[       0.524][110580/168148 34][    0/    0  0] DBUG: Started @ 115200 bps
#[       0.536][110580/168148 34][    0/    0  0] DBUG: Environment: m5stack-core2
#[       0.548][110580/168148 34][    0/    0  0] UART: Started
#[       0.559][110580/168148 34][    0/    0  0] UART: Client login from serial
#[       0.571][110580/168148 34][    0/    0  0] CONF: Failed to load /config.json
Prompt > [       0.584][110580/168148 34][    0/    0  0] CONF: Loading EEPROM
Prompt > [       0.596][110580/168148 34][    0/    0  0] CONF: Loaded EEPROM
Prompt > [       0.608][110580/168208 34][    0/    0  0] DBUG: Loading debug settings
Prompt > [       0.621][110580/168208 34][    0/    0  0] GPIO: Loading GUI settings
Prompt > [       0.634][110580/168208 34][    0/    0  0] HASP: Loading HASP settings
Prompt > [       0.646][110580/168208 34][    0/    0  0] WIFI: Loading WiFi settings
Prompt > [       0.659][110580/168208 34][    0/    0  0] MQTT: Loading MQTT settings
Prompt > [       0.672][110580/168208 34][    0/    0  0] TELN: Loading Telnet settings
Prompt > [       0.685][110580/168208 34][    0/    0  0] MDNS: Loading MDNS settings
Prompt > [       0.698][110580/168208 34][    0/    0  0] HTTP: Loading HTTP settings
Prompt > [       0.710][110580/168208 34][    0/    0  0] GPIO: Loading GPIO settings
Prompt > [       0.723][110580/168208 34][    0/    0  0] CONF: Settings loaded
Prompt > [       0.735][110580/169756 34][    0/    0  0] MSGR: Starting...
Prompt > [       0.747][110580/169756 34][    0/    0  0] MSGR: Started
Prompt > [       0.758][110580/169756 34][    0/    0  0] TFT : Starting...
Prompt > [       1.076][110580/169632 34][    0/    0  0] TFT : Started
Prompt > [       1.480][110580/169536 34][    0/    0  0] LVGL: Starting...
Prompt > [       1.499][110580/169344 34][62320/62544  1] LVGL: LV_FS_PC is initialized with.
Prompt > [       1.514][110580/169344 34][62320/62544  1] HASP: TEST Opening config.json from FS failed 12
Prompt > [       1.532][86004/144188 40][61620/61668  1] GUI : Initialize Cursor
Prompt > [       1.546][86004/144188 40][61528/61552  1] DRVR: Touch SDA     : 21
Prompt > [       1.558][86004/144188 40][61528/61552  1] DRVR: Touch SCL     : 22
Prompt > [       1.570][86004/144188 40][61528/61552  1] DRVR: Touch freq.   : 400000
Prompt > [       1.583][86004/144188 40][61528/61552  1] DRVR: Touch address : 38

**Comment by fvanroie** [comment]
The Tough uses some of the same components as the Core2, but not everything is connected in the same way. So just flashing the Core2 binary onto the Tough will not work. I think the backlight pin is connected via a different port to the AXP192 power management chip, so it could just be a backlight thing...

I haven't looked deeply into the Tough hardware, but I'm aware it uses a more exotic touch panel which will require a new driver for it.
It's not possible to give any coding advice without having hands-on experience with the device... I would first look into the AXP192  differences between the devices to get the backlight to turn on.

**Reply by ritterr** [reply]
My coding experience would eventually not allow me to progress on this topic, as mentioned I would be more than happy to donate a device as such. Can we get in touch on PM?

**Reply by fvanroie** [reply]
Feel free to hop on our Discord server and send a PM.

**Comment by 303Bryan** [comment]
I just picked up one of these slick units. Let me know if you need help testing; I would love to integrate it with HA.

**Reply by 303Bryan** [reply]
@ritterr - I'm curious if you've had any luck getting the M5 Tough working?

**Reply by fvanroie** [reply]
No I haven't, maybe @ritterr did? Feel free to share your configuration here if you make any progress.

**Reply by ritterr** [reply]
Unfortunately I had to put that project on hold. I will revert once I have made some progress
